src/datadog/grok/parse_grok.rs

```diff
@@ -51,19 +51,15 @@ fn apply_grok_rule(source: &str, grok_rule: &GrokRule) -> Result<Value, Error> {
                     filters,
                 }) = grok_rule.fields.get(name)
                 {
-                    filters.iter().for_each(|filter| {
-                    if let Some(ref mut v) = value {
-                        value = match apply_filter(v, filter) {
-                            Ok(Value::Null) => None,
-                            Ok(v) if v.is_object() => Some(parse_keys_as_path(v)),
-                            Ok(v) => Some(v),
-                            Err(error) => {
-                                warn!(message = "Error applying filter", field = %field, filter = %filter, %error);
-                                None
-                            }
-                        };
+                    for filter in filters {
+                        if let Some(ref mut v) = value {
+                            value = match apply_filter(v, filter)? {
+                                Value::Null => None,
+                                v if v.is_object() => Some(parse_keys_as_path(v)),
```

Comment:
I know this is copied code, but I think this could be written as the following to match the above:
```suggestion
                                Value::Object(_) => Some(parse_keys_as_path(v)),
```

Reply:
I looked into it but changing the declaration like that doesnt work because the parse_keys_as_path expects to take ownership of the enum Value, not a variant of it.